libs/testserver/fake_workspace.go

```diff
@@ -227,6 +228,16 @@ func (s *FakeWorkspace) PipelinesCreate(r pipelines.PipelineSpec) Response {
 	}
 }
 
+func (s *FakeWorkspace) PipelinesDelete(pipelineId string) Response {
```

Comment:
Is MapDelete not suitable for this?

Reply:
I missed that this exists. works perfectly fine - see the follow up commit